### PR TITLE
Convert to logxi

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

File: builtin/credential/ldap/backend.go [hunk 1]
```diff
@@ -281,18 +291,20 @@ func (b *backend) getLdapGroups(cfg *ConfigEntry, c *ldap.Conn, userDN string, u
 	ldapMap := make(map[string]bool)
 
 	if cfg.GroupFilter == "" {
-		b.Logger().Printf("[WARN] auth/ldap: GroupFilter is empty, will not query server")
+		b.Logger().Warn("auth/ldap: GroupFilter is empty, will not query server")
```
Comment (vishalnayak) [inline builtin/credential/ldap/backend.go]: IsWarn check is missing here and in the next log message as well.

As a general comment, it is a real pain to check IsDebug or IsWarn before printing anything.

Is it not possible that the Logger().Debug() will only print if Debug is enabled. So we can conveniently skip the check.

File: command/server.go [hunk 2]
```diff
@@ -73,6 +75,38 @@ func (c *ServerCommand) Run(args []string) int {
...
+		level = log.LevelNotice
+	case "warn":
+		level = log.LevelWarn
+	case "err":
+		level = log.LevelError
+	default:
+		c.Ui.Output(fmt.Sprintf("Unknown log level %s", logLevel))
+		return 1
+	}
+	switch os.Getenv("LOGXI_FORMAT") {
```
Comment (vishalnayak) [inline command/server.go]: Should we rename this to VAULT_LOGXI_FORMAT?

File: command/server.go [hunk 3]
```diff
@@ -73,6 +75,38 @@ func (c *ServerCommand) Run(args []string) int {
 		return 1
 	}
 
+	// Create a logger. We wrap it in a gated writer so that it doesn't
+	// start logging too early.
+	logGate := &gatedwriter.Writer{Writer: colorable.NewColorable(os.Stderr)}
+	var level int
+	switch logLevel {
```
Comment (vishalnayak) [inline command/server.go]: Should we do a `strings.ToLower` here?

File: builtin/credential/ldap/backend.go (hunk 1 above)
Comment (jefferai) [inline builtin/credential/ldap/backend.go]: Discussed in Slack; saves the cost of creating arguments. So it's only
used when there are arguments, and optional even then (but good).

File: command/server.go (hunk 2 above)
Comment (jefferai) [inline command/server.go]: Right now this isn't really exposed, but the nice thing about leaving
it as-is is that people can use normal logxi formatters too.

File: command/server.go (hunk 3 above)
Comment (jefferai) [inline command/server.go]: I added this check to `command/server.go`.

File: helper/logformat/vault.go [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+		writer.Write([]byte(" [NOT] "))
+	case log.LevelDebug:
+		writer.Write([]byte(" [DBG] "))
+	case log.LevelTrace:
+		writer.Write([]byte(" [TRC] "))
+	default:
+		writer.Write([]byte(" [ALL] "))
+	}
+
+	writer.Write([]byte(msg + " "))
```
Comment (vishalnayak) [inline helper/logformat/vault.go]: If we remove the " " from here, we can remove the `if i > 0 {` check below?

**Comment by vishalnayak** [comment]
LGTM! Very good job!

File: helper/logformat/vault.go (hunk 4 above)
Comment (jefferai) [inline helper/logformat/vault.go]: Good point -- can't remove here, but can remove the one after msg.

**Pull request merged**